src/java/org/apache/cassandra/io/tries/Walker.java

```diff
@@ -0,0 +1,381 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.io.tries;
+
+import java.io.PrintStream;
+import java.nio.ByteBuffer;
+
+import org.apache.cassandra.io.util.Rebufferer;
+import org.apache.cassandra.io.util.Rebufferer.BufferHolder;
+import org.apache.cassandra.utils.PageAware;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.cassandra.utils.bytecomparable.ByteSource;
+
+/**
+ * Thread-unsafe trie walking helper. This is analogous to RandomAccessReader for tries -- takes an on-disk trie
+ * accessible via a supplied Rebufferer and lets user seek to nodes and work with them.
+ * <p>
+ * Assumes data was written using page-aware builder and thus no node crosses a page and thus a buffer boundary.
+ */
+// TODO STAR-247: unit test are insufficient - they did not catch a problem fixed in STAR-247
```

Comment:
We shouldn't reference STAR tickets.

Reply:
This class has a bunch of unused methods, around half of it is never used.

Reply:
This is related to a comment above. The trie tests were missing that were using these methods. As an aside, I have not marked the methods that are only used by tests as `@VisibleForTesting` because this is a utility class that may be used for tooling in the future. If this isn't a valid reason then I can add the annotation,